pyteal/ast/gtxn_test.py

```diff
@@ -675,6 +675,19 @@ def test_gtxn_id_dynamic():
     assert actual == expected
 
 
+def test_gtxn_nonparticipation():
```

Comment:
Could you also add a test called `test_gtxn_nonparticipation_dynamic` which tests the same thing as this, but uses a PyTeal expression (e.g. `Int(0)`) as the txn index? There are a bunch of tests like that in this file too.

Additionally, once you change the minimum TEAL version to 5, can you add a check to verify compiling with an older version of TEAL raises an error? See `test_txn_extra_program_pages` for an example